plugins/steve/resource-class.ts

```diff
@@ -687,20 +730,20 @@ export default class Resource {
   }
 
   get canEditYaml() {
-    return this.schema?.resourceMethods?.find(x => x === 'blocked-PUT') ? false : this.canUpdate;
+    return this.schema?.resourceMethods?.find((x: string) => x === 'blocked-PUT') ? false : this.canUpdate;
   }
 
   // ------------------------------------------------------------------
 
-  hasLink(linkName) {
+  hasLink(linkName: string) {
     return !!this.linkFor(linkName);
   }
 
-  linkFor(linkName) {
+  linkFor(linkName: string) {
     return (this.links || {})[linkName];
   }
 
-  followLink(linkName, opt = {}) {
+  followLink(linkName: string, opt = {} as HttpRequest) {
```

Comment:
The same kind of thing applies here with the `as`s. 

```
followLink(linkName: string, opt = {} as HttpRequest) {

// to

followLink(linkName: string, opt: HttpRequest = {}) {

```

Reply:
Changed it to the latter